### ISSUE TITLE
Unable to deploy Python closure: hangs indefinitely on cloudpickle error

### ISSUE THREAD

**Issue by TomNash** [ucbrise/clipper#568]
Following the steps for [XGBoost deployment](https://clipper.ai/tutorials/xgboost_deployment/), I run into an issue with the command

```
python_deployer.deploy_python_closure(cl, name='xgboost-model', version=1,
     input_type="integers", func=predict, pkgs_to_install=['xgboost'])
```

The same behavior occurs with the [quickstart guide](http://clipper.ai/), running

```
python_deployer.deploy_python_closure(clipper_conn, name="sum-model", version=1, input_type="doubles", func=feature_sum)
```

It seems this `while` loop in [docker_container_manager.py](https://github.com/ucbrise/clipper/blob/develop/clipper_admin/clipper_admin/docker/docker_container_manager.py#L362) will loop infinitely with no error if the container creation fails.

```
> docker inspect 3dbc59e5619d
[
    {
        "Id": "3dbc59e5619d5f0445cf4b85c3892aad7987ce7bb4a7946a82c89e3a2f440bbf",
        "Created": "2018-09-11T20:31:56.5002835Z",
        "Path": "/container/container_entry.sh",
        "Args": [
            "py-closure-container",
            "/container/python_closure_container.py"
        ],
        "State": {
            "Status": "exited",
            "Running": false,
...
```

I checked the log of the container to see this

```
> docker container logs 3dbc59e5619d
Starting Python Closure container
Connecting to Clipper with default port: 7000
Traceback (most recent call last):
  File "/container/python_closure_container.py", line 56, in <module>
    rpc_service.get_input_type())
  File "/container/python_closure_container.py", line 28, in __init__
    self.predict_func = load_predict_func(predict_path)
  File "/container/python_closure_container.py", line 17, in load_predict_func
    return cloudpickle.load(serialized_func_file)
  File "/usr/local/lib/python3.6/site-packages/cloudpickle/cloudpickle.py", line 1060, in _make_skel_func
    base_globals['__builtins__'] = __builtins__
TypeError: 'str' object does not support item assignment
Encountered error not related to missing packages. Please refer to the container log to diagnose.
```

Any advice on how to proceed?

**Comment by udaynaik** [comment]
I have same issue using the example from quickstart. 

18-09-11:16:59:37 INFO     [clipper_admin.py:201] Application TestApp was successfully registered
18-09-11:16:59:37 INFO     [deployer_utils.py:44] Saving function to /tmp/clipper/tmp034h3wv3
18-09-11:16:59:37 INFO     [deployer_utils.py:54] Serialized and supplied predict function
18-09-11:16:59:37 INFO     [python.py:192] Python closure saved
18-09-11:16:59:37 INFO     [python.py:206] Using Python 3.6 base image
18-09-11:16:59:37 INFO     [clipper_admin.py:452] Building model Docker image with model data from /tmp/clipper/tmp034h3wv3
18-09-11:16:59:37 INFO     [clipper_admin.py:456] {'stream': 'Step 1/2 : FROM clipper/python36-closure-container:0.3.0'}
18-09-11:16:59:37 INFO     [clipper_admin.py:456] {'stream': '\n'}
18-09-11:16:59:37 INFO     [clipper_admin.py:456] {'stream': ' ---> 74ba26b9a6ba\n'}
18-09-11:16:59:37 INFO     [clipper_admin.py:456] {'stream': 'Step 2/2 : COPY /tmp/clipper/tmp034h3wv3 /model/'}
18-09-11:16:59:37 INFO     [clipper_admin.py:456] {'stream': '\n'}
18-09-11:16:59:37 INFO     [clipper_admin.py:456] {'stream': ' ---> 07a12e257348\n'}
18-09-11:16:59:37 INFO     [clipper_admin.py:456] {'aux': {'ID': 'sha256:07a12e2573483741b5ac225b509880210b1b202553b67f1d05200b443e68da9a'}}
18-09-11:16:59:37 INFO     [clipper_admin.py:456] {'stream': 'Successfully built 07a12e257348\n'}
18-09-11:16:59:37 INFO     [clipper_admin.py:456] {'stream': 'Successfully tagged sum-model:1\n'}
18-09-11:16:59:37 INFO     [clipper_admin.py:458] Pushing model Docker image to sum-model:1
18-09-11:16:59:39 INFO     [docker_container_manager.py:257] Found 0 replicas for sum-model:1. Adding 1

---hung----

My versions:
Mac : 10.12.6
Docker Version 18.03.0-ce-mac60 (23751)

clipper-admin             0.3.0    
cloudpickle                0.5.3

**Comment by simon-mo** [comment]
The quick fix is to `pip install cloudpickle==0.5.3` the reason is that your local cloudpickle version is higher than the one used in docker container, making them incompatible.
